lib/client.js

```diff
@@ -1278,6 +1280,132 @@ function _resume (client, sync) {
   }
 }
 
+async function writeIterable (client, request, socket, contentLength, header, expectsPayload) {
+  const { body } = request
+  socket[kWriting] = true
+  assert(contentLength !== 0 || client[kRunning] === 0, 'stream body cannot be pipelined')
+
+  let finished = false
+  let bytesWritten = 0
+  function onFinished (err) {
+    if (finished) {
+      return
+    }
+
+    finished = true
+    socket[kWriting] = false
+    socket.removeListener('error', onFinished)
+    finishChunkedWrite(err, contentLength, bytesWritten, expectsPayload, header, socket, client)
+  }
+
+  let thrownError = null
+  try {
+    socket.on('error', onFinished)
```

Comment:
I wasn't sure if this is strictly needed, however there's similar logic in the stream path